### PR TITLE
PP-570/pricer-case-sensitive

### PR DESCRIPTION
**Pull request by franciscotobar**
## What

- The relayPricer and API needs to handle case sensitive Currencies and Tokens. 
- Typing to guarantee the appropriate name.

## Why

- There are different type of casing for tokens and currencies and we need to be able to let the programmer know that the name that is using is valid.

## Refs

- (PP-570)[https://rsklabs.atlassian.net/browse/PP-570]

### CODE REVIEW

File: src/api/ExchangeApi.ts [hunk 1]
```diff
@@ -1,26 +1,39 @@
...
                 `${this.api} API cannot map a token with a null/empty value`
             );
         }
+
         const upperCaseTokenSymbol = tokenSymbol.toUpperCase();
-        const resultMapping = this.currencyMapping[upperCaseTokenSymbol];
-        if (!resultMapping) {
+
+        const currency =
+            this.currencyMapping[upperCaseTokenSymbol as FromCurrency];
```
Comment (ironFe93) [inline src/api/ExchangeApi.ts:28]: would try to avoid the 'as FromCurrency' part but I'm probably being too nitpicky

**Review by ironFe93 — APPROVED** [review]
lgtm

**Comment by franciscotobar** [comment]
you are right, I tried to avoid it several times but I need an opinion about it, maybe we can discuss it but if you see the history, I did a commit without forcing the typing but I change it.

**Review by antomor — APPROVED** [review]
LGTM

**Pull request merged**